pkg/bath/straws.go

```diff
@@ -0,0 +1,194 @@
+			recipient: parseAccount(transfer.NewOwner),
+		},
+		Accounts: append(bubble.Accounts, nftBubble.account.Address),
+		Fee:      bubble.Fee,
+	}
+	newBubble.Fee.WhoPay = nftBubble.inputFrom.Address
+	newBubble.Fee.Deposit += nftBubble.inputAmount
+	newBubble.Children = processChildren(bubble.Children,
+		func(tx BubbleTx, child *Bubble) *merge {
+			if !tx.operation("Excess") {
+				return nil
+			}
+			newBubble.Fee.Add(child.Fee)
+			newBubble.Fee.Refund += tx.inputAmount
+			newBubble.Accounts = append(newBubble.Accounts, tx.account.Address)
+			return &merge{children: child.Children}
+		},
+		func(tx BubbleTx, child *Bubble) *merge {
+			if !tx.operation("NftOwnershipAssigned") {
+				return nil
+			}
+			newBubble.Fee.Add(child.Fee)
+			newBubble.Accounts = append(newBubble.Accounts, tx.account.Address)
+			return &merge{children: child.Children}
+		})
+	*bubble = newBubble
+	return true
+}
+
+type BubbleNftTransfer struct {
+	success   bool
+	account   Account
+	sender    *Account
+	recipient *Account
+	payload   any //todo: replace any
+}
+
+func (b BubbleNftTransfer) ToAction() (action *Action) {
+	return &Action{
+		NftItemTransfer: &NftTransferAction{
+			Comment:   nil, //todo: add
+			Recipient: b.account.Addr(),
+			Sender:    b.sender.Addr(),
+			Nft:       b.account.Address,
+		},
+		Success: b.success,
+		Type:    NftItemTransfer,
+	}
+}
+
+func FindJettonTransfer(bubble *Bubble) bool {
+	jettonBubble, ok := bubble.Info.(BubbleTx)
+	if !ok {
+		return false
+	}
+	if !jettonBubble.operation("JettonTransfer") {
+		return false
+	}
+	intention := jettonBubble.decodedBody.Value.(abi.JettonTransferMsgBody)
+	transfer := BubbleJettonTransfer{
+		sender:       jettonBubble.inputFrom,
+		senderWallet: jettonBubble.account.Address,
+		master:       tongo.AccountID{},
+		amount:       intention.Amount,
+		payload:      nil, //todo: do
+	}
+	newBubble := Bubble{
+		Accounts: append(bubble.Accounts, jettonBubble.account.Address),
+		Children: bubble.Children,
+		Fee:      bubble.Fee,
+	}
+	if jettonBubble.success {
+		newBubble.Children = processChildren(bubble.Children,
+			func(tx BubbleTx, child *Bubble) *merge {
+				if !tx.operation("JettonInternalTransfer") {
+					return nil
+				}
+				children := processChildren(child.Children,
+					func(tx BubbleTx, excess *Bubble) *merge {
+						if !tx.operation("Excess") {
+							return nil
+						}
+						return &merge{children: excess.Children}
+					},
+					func(tx BubbleTx, notify *Bubble) *merge {
+						if !tx.operation("JettonNotify") {
+							return nil
+						}
+						transfer.recipient = g.Pointer(tx.account)
+						return &merge{children: notify.Children}
+					},
+				)
+				return &merge{children: children}
+			})
+		if transfer.recipient == nil {
+			transfer.recipient = parseAccount(intention.Destination)
+		}
+	}
+	newBubble.Info = transfer
+	newBubble.Fee.WhoPay = jettonBubble.inputFrom.Address
+	*bubble = newBubble
+	return true
+}
+
+type BubbleJettonTransfer struct {
```

Comment:
how about not exporting this struct? 
`bubbleJettonTransfer` or even `jettonTransfer` should work